### PR TITLE
Compile used contracts

### LINKED ISSUE #406: Verify if user can compile contract from a dependency in the project
Use-case is deploying those contracts using protostar
It should be done by adding something like this to `lib.cairo`
```
use my_contract::ContractA;
```

### PR DESCRIPTION
**Pull request by maciektr**
fix #406 
CC @MaksymilianDemitraszek

### CODE REVIEW

File: extensions/scarb-cairo-test/Cargo.toml [hunk 1]
```diff
@@ -16,6 +16,7 @@ clap.workspace = true
 scarb-metadata = { path = "../../scarb-metadata", features = ["packages_filter"] }
 serde_json.workspace = true
 
+cairo-lang-utils.workspace = true
```
Comment (mkaput) [inline extensions/scarb-cairo-test/Cargo.toml:19]: this line seems to be put off

File: scarb/src/compiler/compilers/starknet_contract.rs [hunk 2]
```diff
@@ -213,6 +221,69 @@ impl Compiler for StarknetContractCompiler {
...
+            if let ModuleId::Submodule(submodule_id) = module_id {
+                contracts.push(ContractDeclaration { submodule_id });
+            } else {
+                panic!("Contract `{contract_name}` was not found.");
+            }
+        }
+    }
+    contracts
+}
+
```
Comment (mkaput) [inline scarb/src/compiler/compilers/starknet_contract.rs:286]: could we upstream these changes to the compiler itself?

**Review by mkaput — CHANGES_REQUESTED** [review]

File: scarb/src/compiler/compilers/starknet_contract.rs (hunk 2 above)
Comment (maciektr) [reply scarb/src/compiler/compilers/starknet_contract.rs:286]: Sure! Do you think we can upstream whole logic of searching imported modules, or only the `find_module_contracts` function that we can reuse?

Comment (mkaput) [reply scarb/src/compiler/compilers/starknet_contract.rs:286]: whole logic. I think `starknet-compile` should also behave like this for consistency

**Comment by maciektr** [comment]
superseded by https://github.com/software-mansion/scarb/pull/458

**Comment by maciektr** [comment]
superseded by https://github.com/software-mansion/scarb/pull/458

**Pull request closed without merging**